src/Mod/Fem/App/FemMesh.cpp

```diff
@@ -971,25 +971,37 @@ std::set<int> FemMesh::getNodesByEdge(const TopoDS_Edge &edge) const
     // get the current transform of the FemMesh
     const Base::Matrix4D Mtrx(getTransform());
 
+    std::vector<const SMDS_MeshNode*> nodes;
     SMDS_NodeIteratorPtr aNodeIter = myMesh->GetMeshDS()->nodesIterator();
     while (aNodeIter->more()) {
         const SMDS_MeshNode* aNode = aNodeIter->next();
-        Base::Vector3d vec(aNode->X(),aNode->Y(),aNode->Z());
+        nodes.push_back(aNode);
+    }
+
+#pragma omp parallel for schedule(dynamic)
```

Comment:
The #pragma tag is new for me. What it is for/how does it work?

Reply:
I used it following the instructions to run openmp, which is for thread level parallelization. The line instructs the program to generate threads as available and have them take care of one iterator of the for loop per thread. Any variables declared inside the loop are local to each thread, but others are shared between all threads.

As for the `#pragma` directive itself, I do not have an exact answer for what it does. The best I could find is that it "is the method specified by the C standard for providing additional information to the compiler, beyond what is conveyed in the language itself" (https://gcc.gnu.org/onlinedocs/cpp/Pragmas.html).

Reply:
OK, then please rebase and I will try what the MSVC compiler telly me about this.

Reply:
Rebased.